_code-samples/quickstart/js/ripplex3a-create-offers.js

```diff
@@ -0,0 +1,276 @@
+  const signed = standby_wallet.sign(prepared)
+
+  // -------------------------------------------------------- Submit signed blob
+  const tx = await client.submitAndWait(signed.tx_blob)
+      
+  results  += "\nBalance changes: \n" + 
+    JSON.stringify(xrpl.getBalanceChanges(tx.result.meta), null, 2)
+  standbyResultField.value = results
+
+  standbyBalanceField.value =  (await client.getXrpBalance(standby_wallet.address))
+  client.disconnect()    
+  } // End of cancelOffer()
+
+  /*********************************************
+  ************* Reciprocal Functions **********
+  ********************************************/
+
+  /***********************************
+   ********* OP Create Offer *********
+    **********************************/
+
+    async function oPcreateOffer() {
+    let takerGets = ''
+    let takerPays = ''
+
+      operationalResultField.value = ''
+      let net = getNet()
+      let results = 'Connecting to ' + net + '....\n'
+      const client = new xrpl.Client(net)
+      await client.connect()
+          
+      results  += "Connected. Getting wallets.\n"
+      operationalResultField.value = results
+      const standby_wallet = xrpl.Wallet.fromSeed(standbySeedField.value)
+      const operational_wallet = xrpl.Wallet.fromSeed(operationalSeedField.value)
+      results += operationalNameField.value + " account address: " + operational_wallet.address + "\n"
+      operationalResultField.value = results
+
+
+    if (operationalTakerGetsCurrencyField.value == 'XRP') {
+      takerGets = operationalTakerGetsValueField.value
+    } else {
+      takerGetsString = '{"currency": "' + operationalTakerGetsCurrencyField.value +'",\n' +
+          '"issuer": "' + operationalTakerGetsIssuerField.value + '",\n' +
+          '"value": "' + operationalTakerGetsValueField.value + '"}'
+      takerGets = JSON.parse(takerGetsString)
+    }
+
+    if (operationalTakerPaysCurrencyField.value == 'XRP') {
+      takerPays = operationalTakerPaysValueField.value
+    } else {
+      takerPaysString = '{"currency": "' + operationalTakerPaysCurrencyField.value + '",\n' +
+        '"issuer": "' + operationalTakerPaysIssuerField.value + '",\n' +
+        '"value": "' + operationalTakerPaysValueField.value + '"}'
+      takerPays = JSON.parse(takerPaysString)
+    }
+    
+      // -------------------------------------------------------- Prepare transaction
+    const prepared = await client.autofill({
+      "TransactionType": "OfferCreate",
+      "Account": operational_wallet.address,
+      "TakerGets": takerGets,
+        "TakerPays": takerPays
+    })   
+    // ------------------------------------------------- Sign prepared instructions
+    const signed = operational_wallet.sign(prepared)
+    results += "\nSubmitting transaction...."
+    // -------------------------------------------------------- Submit signed blob
+    const tx = await client.submitAndWait(signed.tx_blob)
+    
+    results += tx.results + "\n"
```

Comment:
The operational log is showing `undefined` when you submit an offer create transaction. Looks like something is going on here.

Reply:
Yep, some dev-time kruft that shouldn't have survived this long. Quashed that and a couple of typos. Thanks for pointing that out.